code/modules/food_and_drinks/kitchen_machinery/gibber.dm

```diff
@@ -195,16 +195,13 @@
 
 	if(typeofskin)
 		skin = new typeofskin
-
 	log_combat(user, occupant, "gibbed")
 	mob_occupant.investigate_log("has been gibbed by [src].", INVESTIGATE_DEATHS)
 	mob_occupant.death(TRUE)
 	mob_occupant.ghostize()
-	set_occupant(null)
-	qdel(mob_occupant)
-	addtimer(CALLBACK(src, PROC_REF(make_meat), skin, allmeat, meat_produced, gibtype, diseases), gibtime)
+	addtimer(CALLBACK(src, PROC_REF(make_meat), skin, allmeat, meat_produced, gibtype, diseases, mob_occupant), gibtime)
 
-/obj/machinery/gibber/proc/make_meat(obj/item/stack/sheet/animalhide/skin, list/obj/item/reagent_containers/food/snacks/meat/slab/allmeat, meat_produced, gibtype, list/datum/disease/diseases)
+/obj/machinery/gibber/proc/make_meat(obj/item/stack/sheet/animalhide/skin, list/obj/item/reagent_containers/food/snacks/meat/slab/allmeat, meat_produced, gibtype, list/datum/disease/diseases, mob_occupant)
```

Comment:
Since it's addtimer(), It's possible to see a short time gap to cause a glitch. There should be a check if a mob is still inside of the gibber.
This should be done because `qdel(mob)` was moved to after addtimer().

Reply:
👍